hydrosdk/model.py

```diff
@@ -207,11 +212,17 @@ def upload(self, cluster) -> dict:
                 msc = MetricSpecConfig(model_version_id=upload_response.model_version_id,
                                        threshold=metric.threshold,
                                        threshold_op=metric.comparator)
-
-                ms = MetricSpec.create(cluster=upload_response.model.cluster,
-                                       name=upload_response.model.name,
-                                       model_version_id=root_model_upload_response.model_version_id,
-                                       config=msc)
+                ms_created = False
+
+                while not ms_created:
+                    try:
+                        ms = MetricSpec.create(cluster=upload_response.model.cluster,
+                                               name=upload_response.model.name,
+                                               model_version_id=root_model_upload_response.model_version_id,
+                                               config=msc)
+                        ms_created = True
+                    except Exception:
```

Comment:
Is there a specific Exception that you wait for, or sleep on everything?
if there is a specific exception, then it would be better to narrow down the Exception type

Reply:
Fixed